client/logic/game/terrain/terrain.gd

```diff
@@ -97,6 +99,21 @@ func uncapture() -> void:
 	stats.reset_capture_health()
 
 
+func get_terrain_by_position(pos: Vector2i) -> Terrain:
+	# check if terrain lookup is generated (dictionary exist and entries aren't null). If not -> generate
+	if not (_terrain_lookup and _terrain_lookup[_terrain_lookup.keys()[0]] and is_instance_valid(_terrain_lookup[_terrain_lookup.keys()[0]])):
```

Comment:
is it not better, to initializate the terrain_lookup to an empty dictionary, and then check that the number of elements is greater than 0?

Reply:
The problem is, that the dictionary can be already filled by the last game (and even when looking on a map with the map selection). Otherwise `if not _terrain_lookup` would be enough.

Reply:
maybe because last years I was developing react and this kind of stuff, but maybe the solution is to ensure the terrain is allways new?

Reply:
Creating all the time the look up dictionary would make it less effective.